syncmaster/schemas/v1/connections/hdfs.py

```diff
@@ -1,40 +1,62 @@
 # SPDX-FileCopyrightText: 2023-2024 MTS PJSC
 # SPDX-License-Identifier: Apache-2.0
 
-from pydantic import BaseModel, SecretStr
+from pydantic import BaseModel, Field
 
+from syncmaster.schemas.v1.auth import (
+    CreateBasicAuthSchema,
+    ReadBasicAuthSchema,
+    UpdateBasicAuthSchema,
+)
 from syncmaster.schemas.v1.connection_types import HDFS_TYPE
+from syncmaster.schemas.v1.connections.connection_base import (
+    CreateConnectionBaseSchema,
+    ReadConnectionBaseSchema,
+    UpdateConnectionBaseSchema,
+)
 
 
 class HDFSConnectionBaseSchema(BaseModel):
-    type: HDFS_TYPE
 
     class Config:
         from_attributes = True
 
 
-class HDFSReadConnectionSchema(HDFSConnectionBaseSchema):
+class CreateHDFSConnectionDataSchema(HDFSConnectionBaseSchema):
     cluster: str
 
 
-class HDFSCreateConnectionSchema(HDFSConnectionBaseSchema):
+class ReadHDFSConnectionDataSchema(HDFSConnectionBaseSchema):
     cluster: str
 
 
-class HDFSUpdateConnectionSchema(HDFSConnectionBaseSchema):
+class UpdateHDFSConnectionDataSchema(HDFSConnectionBaseSchema):
     cluster: str
 
 
-class HDFSReadAuthSchema(HDFSConnectionBaseSchema):
-    user: str
+class CreateHDFSConnectionSchema(CreateConnectionBaseSchema):
+    type: HDFS_TYPE = Field(..., description="Connection type")
+    data: CreateHDFSConnectionDataSchema = Field(
+        ...,
+        alias="connection_data",
+        description=(
+            "Data required to connect to the database. These are the parameters that are specified in the URL request."
+        ),
+    )
+    auth_data: CreateBasicAuthSchema = Field(
+        ...,
+        discriminator="type",
```

Comment:
Actually, as there is no Union, you can drop `discriminator` and `...` from all ConnectionSchema